### PR TITLE
EVAKA-4481 Associate messages with children

### PR DESCRIPTION
**Pull request by zch**
Help the employees identify who a message is regarding by allowing the guardian to indicate who they are messaging about.

Citizen composing a message:
image

Unit supervisor's inbox:
image

### CODE REVIEW

File: frontend/src/citizen-frontend/messages/MessageEditor.tsx [hunk 1]
```diff
@@ -105,6 +141,41 @@ export default React.memo(function MessageEditor({
               />
             </label>
 
+            <Gap size="s" />
+            <label>
+              <Bold>{i18n.messages.messageEditor.children}</Bold>
+              {renderResult(children, (children) => (
+                <FixedSpaceFlexWrap horizontalSpacing="xs">
+                  {childIds &&
+                    childIds.length > 1 &&
```
Comment (jaahon) [inline frontend/src/citizen-frontend/messages/MessageEditor.tsx:150]: This condition should be outside the label as well.

**Review by jaahon — COMMENTED** [review]
When citizens reply to a message sent to them there's no child selection and the employees do not see whose parent is sending the message.

Also if the citizen's children are in multiple daycares and a message is simultaneously sent to groups in different units all of the children are shown in the respective group's accounts. It might not be a huge issue but it still leaks personal data that the recipients probably should not get.

**Comment by zch** [comment]
> When citizens reply to a message sent to them there's no child selection and the employees do not see whose parent is sending the message.

Children are now automatically added to the messages created by staff so the names of the children involved are shown to the staff members when they read the messages, but not to the parent (yet. Not sure if necessary?).

> Also if the citizen's children are in multiple daycares and a message is simultaneously sent to groups in different units all of the children are shown in the respective group's accounts. It might not be a huge issue but it still leaks personal data that the recipients probably should not get.

Interesting point, have to think about this a bit. I'm trying to think of relevant cases where I, as a parent, would want to send the same message to different daycares regarding different children. 🤔

**Comment by jaahon** [comment]
> > Also if the citizen's children are in multiple daycares and a message is simultaneously sent to groups in different units all of the children are shown in the respective group's accounts. It might not be a huge issue but it still leaks personal data that the recipients probably should not get.
> 
> Interesting point, have to think about this a bit. I'm trying to think of relevant cases where I, as a parent, would want to send the same message to different daycares regarding different children. 🤔

It's probably not likely nor intended but it's still possible. Just a concern I had.

**Review by jaahon — APPROVED** [review]
Looks good to me

**Pull request merged**